packages/syntax-core/src/TapArea/TapArea.test.tsx

```diff
@@ -0,0 +1,170 @@
+import { screen, render } from "@testing-library/react";
+import TapArea from "./TapArea";
+import { expect, vi } from "vitest";
+import userEvent from "@testing-library/user-event";
+import { createRef } from "react";
+
+describe("tapArea", () => {
+  it("renders successfully", () => {
+    const { baseElement } = render(
+      <TapArea
+        onTap={() => {
+          /* empty */
+        }}
+      />,
+    );
+    expect(baseElement).toBeTruthy();
+  });
+
+  it("renders an role=button element", async () => {
+    render(
+      <TapArea
+        onTap={() => {
+          /* empty */
+        }}
+      />,
+    );
+    const tapArea = await screen.findAllByRole("button");
+    expect(tapArea).toHaveLength(1);
+  });
+
+  it("renders its children", async () => {
+    render(
+      <TapArea
+        onTap={() => {
+          /* empty */
+        }}
+      >
+        <div data-testid="tap-area-child-testid">Continue</div>
+      </TapArea>,
+    );
+    const child = await screen.findByTestId("tap-area-child-testid");
+    expect(child).toHaveTextContent("Continue");
+  });
+
+  it("correctly applies fullWidth when set", async () => {
+    render(
+      <TapArea
+        data-testid="tap-area-testid"
+        onTap={() => {
+          /* empty */
+        }}
+        fullWidth
+      />,
+    );
+    const tapArea = await screen.findByTestId("tap-area-testid");
+    expect(tapArea).toHaveStyle({ width: "100%" });
+  });
+
+  it("sets an accessibility label", async () => {
+    render(
+      <TapArea
+        data-testid="tap-area-testid"
+        onTap={() => {
+          /* empty */
+        }}
+        accessibilityLabel="Continue to the next step"
+        fullWidth
+      />,
+    );
+    const tapArea = await screen.findAllByLabelText(
```

Comment:
I noticed that some places use `findByTestId` and some use `findAllByLabelText`. Do you use `findAllByLabelText` for testing when you need to need something done externally for it?

and `findByTestId`, when you need to see the internals of the component?

Reply:
@somethiiing I used both in these tests just to make sure both worked (`data-testid` and label text).